### DISCUSSION TITLE
Upsert: Only update columns in upsert object

### DISCUSSION THREAD

**Discussion by taylorhlabs** [supabase/supabase#3447] [Questions] [answered]
I am trying to bulk update a 'sort_order' column in a table with 10+ columns using the `upsert` function. It seems that the `upsert` function replaces the entire matching row, but I'd like to update just the columns in the object passed to `upsert`. For instance, I'd like to do something like
```
const { data, error } = await supabase
  .from('my_table')
  .upsert([
   {id: 1, sort_order: 1},
   {id: 2, sort_order: 2}
  ])
```
and only update the sort_order column but leave all other columns in the table as is. Is this possible? Or do I have to pass data for all columns to avoid losing the data in columns that are not sort_order?

**Comment by thebengeu** [comment, marked as answer]
I don't think this is supported, per [these docs](https://postgrest.org/en/v8.0/api.html?highlight=upsert#put), all columns must be specified. @steve-chavez to confirm.

**Reply by steve-chavez** [reply]
> I don't think this is supported, per these docs, all columns must be specified

Yes, correct. A proper bulk update would solve this - tracked on https://github.com/supabase/postgrest-js/issues/174.

**Reply by darora** [reply]
@steve-chavez is this correct? Don't we have a test that explicitly checks that we support this with merge-duplicates?

https://github.com/PostgREST/postgrest/blob/main/test/spec/Feature/UpsertSpec.hs#L35-L46

**Reply by steve-chavez** [reply]
@darora I think I got confused with the docs linked(PUT method). postgrest-js' upsert is using merge-duplicates(POST with header, not PUT), so yes, that should work.

**Comment by mkmoisen** [comment]
@steve-chavez 

Is it possible to do this partial update yet?

I think this is a common case. 

With the current full-update requirement, I have to select all the columns for my table, which I don't need. Then I have to do this update and send all these columns which I don't need. If another user has updated some data between the select and the bulk update of the current user, the current user will overwrite the update.

Thank you.

**Comment by felipenmoura** [comment]
Well, yeah...that feature would be really interesting.
Feels really weird having to select * to be able to perform an `upsert`...I mean, If I have to select every single column, I could simply perform an `update`, right? No need to upsert even exist. Not worth "saving" an `if (id)` in the code.

Just IMHO

**Comment by juanmartinzzz** [comment]
1 year after the previous person posting, I think this still fails to meet expectations -- I thought using `ignoreDuplicates: false` flag would work for this case but it doesn't seem to be the case, the system overrides the row. 

I also think it would be good to support this so I can selectively update only 1 or a couple (but not ALL) columns in a table!

**Comment by bobbybol** [comment]
Is there at least a workaround for this? I have now found this thread in the discussions, going for four years. I have commented at a [corresponding issue](https://github.com/supabase/supabase-js/issues/1653) in the supabase-js repo. I have contacted the helpdesk. I have followed all the corresponding issues in the github repo that is now deprecated. There has to be some minimal solution to this, or at least something in the pipeline. @steve-chavez @darora ?

**Comment by KiddoV** [comment]
Is this feature possible yet? Would be nice for Supabase to just ignore all those unrelated columns in `data`.
